lib/tokenVersions.js

```diff
@@ -0,0 +1,235 @@
+  }
+  if(id !== undefined) {
+    if(!(Number.isInteger(id) && id >= 0)) {
+      throw new TypeError('"id" must be a non-negative integer.');
+    }
+    query['tokenVersion.id'] = id;
+  }
+  if(tokenizerId !== undefined) {
+    if(typeof tokenizerId !== 'string') {
+      throw new TypeError('"tokenizerId" must be a string.');
+    }
+    query['tokenVersion.tokenizerId'] = tokenizerId;
+  }
+  const projection = {_id: 0};
+  const collection = database.collections['tokenization-tokenVersion'];
+  const record = await collection.findOne(query, projection);
+  if(!record) {
+    const details = {
+      httpStatusCode: 404,
+      public: true
+    };
+    throw new BedrockError(
+      'Token version not found.',
+      'NotFoundError', details);
+  }
+  return record;
+}
+
+export async function setOptions({options}) {
+  const collection = database.collections['tokenization-tokenVersionOptions'];
+  const now = Date.now();
+  const tokenVersionOptions = {
+    id: TOKEN_OPTIONS_ID,
+    options
+  };
+  const query = {'tokenVersionOptions.id': TOKEN_OPTIONS_ID};
+  const $set = {tokenVersionOptions, 'meta.updated': now};
+  const upsertOptions = {...database.writeOptions, upsert: true};
+  const result = await collection.updateOne(query, {
+    $set,
+    $setOnInsert: {'meta.created': now}
+  }, upsertOptions);
+  if(result.result.n > 0) {
+    // upserted or modified
+    return true;
+  }
+  // no update, options were identical
+  return false;
+}
+
+export async function getOptions() {
+  const query = {'tokenVersionOptions.id': TOKEN_OPTIONS_ID};
+  const projection = {_id: 0};
+  const collection = database.collections['tokenization-tokenVersionOptions'];
+  const record = await collection.findOne(query, projection);
+  if(!record) {
+    const details = {
+      httpStatusCode: 404,
+      public: true
+    };
+    throw new BedrockError(
+      'Token version options not found.',
+      'NotFoundError', details);
+  }
+  return record;
+}
+
+export async function insertOptions({options}) {
+  const collection = database.collections['tokenization-tokenVersionOptions'];
+  const now = Date.now();
+  const meta = {created: now, updated: now};
+  let record = {
+    meta,
+    tokenVersionOptions: {
+      id: TOKEN_OPTIONS_ID,
+      options
+    }
+  };
+  try {
+    const result = await collection.insertOne(record, database.writeOptions);
+    record = result.ops[0];
+  } catch(e) {
+    if(!database.isDuplicateError(e)) {
+      throw e;
+    }
+    throw new BedrockError(
+      'Duplicate token version options.',
+      'DuplicateError', {
+        public: true,
+        httpStatusCode: 409
+      }, e);
+  }
+  return record;
+}
+
+async function _getNextVersionId() {
+  const projection = {_id: 0, 'tokenVersion.id': 1};
+  const collection = database.collections['tokenization-tokenVersion'];
+  const records = await collection.find(
+    {}, projection).sort({id: -1}).toArray();
```

Comment:
Should the sort key be `'tokenVersion.id'`?
(Also another place `projection` needs fixing.)

Reply:
The code changed and outdated this, but the sort key question still applies.